### ISSUE TITLE
error with tf.app.flags in fully_connected_feed.py

### ISSUE THREAD

**Issue by sergulaydore** [tensorflow/tensorflow#509]
I am getting the following error when I try to use flags. Can you please advise?
Thanks

flags = tf.app.flags
FLAGS = flags.FLAGS
flags.DEFINE_float('learning_rate', 0.01, 'Initial learning rate.')
flags.DEFINE_integer('max_steps', 2000, 'Number of steps to run trainer.')
flags.DEFINE_integer('hidden1', 128, 'Number of units in hidden layer 1.')
flags.DEFINE_integer('hidden2', 32, 'Number of units in hidden layer 2.')
flags.DEFINE_integer('batch_size', 100, 'Batch size.  '
                     'Must divide evenly into the dataset sizes.')
flags.DEFINE_string('train_dir', 'data', 'Directory to put the training data.')
flags.DEFINE_boolean('fake_data', False, 'If true, uses fake data '
                     'for unit testing.')

The error is:

usage: __main__.py [-h] [--learning_rate LEARNING_RATE]
                   [--max_steps MAX_STEPS] [--hidden1 HIDDEN1]
                   [--hidden2 HIDDEN2] [--batch_size BATCH_SIZE]
                   [--train_dir TRAIN_DIR] [--fake_data FAKE_DATA]
                   [--nofake_data]
__main__.py: error: unrecognized arguments: -f /Users/sergulaydore/.ipython/profile_default/security/kernel-3561b41d-e378-459a-9e96-5333ea079712.json --profile-dir /Users/sergulaydore/.ipython/profile_default

**Comment by mrry** [comment]
How are you invoking your script? From the error message, it looks like you are passing `-f` and `--profile-dir` flags, but neither of them is defined in your code.

**Comment by vrv** [comment]
Also, @craigcitro, in case he knows of a way to improve our thin flags wrapper in any way.

**Comment by craigcitro** [comment]
In particular, from those flags you're passing, I'm guessing you're trying to do something related to IPython? If so, it's probably possible, but we'd need to know what you were trying to do.

@vrv one option would be to use the [parse_known_args](https://docs.python.org/2/library/argparse.html#partial-parsing) method in `_flags.py`, which would let us parse a set of args but return the remaining ones for the caller to use as they'd like (eg pass them on to ipython). this would be a break from the python-gflags behavior, but possibly useful.

**Comment by sergulaydore** [comment]
Thank you so much for your answers but I am just learning tensorflow and was trying you run the tutorials. I am just using the script in https://tensorflow.googlesource.com/tensorflow/+/master/tensorflow/examples/tutorials/mnist/fully_connected_feed.py and getting the error above.
Is it possible that there is a bug in the script provided by tensorflow?

**Comment by craigcitro** [comment]
What was the command-line you used to invoke it? (Can you give us a pointer to where in the tutorial you were?)

**Comment by sergulaydore** [comment]
So, I was going through this tutorial: https://www.tensorflow.org/versions/master/tutorials/mnist/tf/index.html#tensorflow-mechanics-101 and I got an error when reading the data which was 

data_sets = input_data.read_data_sets(FLAGS.train_dir, FLAGS.fake_data)

It works fine if I just enter the directory instead of FLAGS.train_dir.

**Comment by craigcitro** [comment]
Oh, I see -- are you just following the tutorial from within IPython?

**Comment by sergulaydore** [comment]
Exactly! Is that wrong? Should I run from Python shell?

**Comment by craigcitro** [comment]
Well, I think it just means that we didn't anticipate that use case, but should have. :wink: As a short-term workaround, running from the python command-line should get you fixed up.

@vrv it looks like we're going to need something like the fix I was talking about above.

**Comment by sergulaydore** [comment]
Indeed it worked in IPython command line :) Thank you very much!

**Comment by vrv** [comment]
I think I fixed this in 1c579361cd1e088dd5e05a394b1561a73e3667ba
